### PR TITLE
Phase 2 of reproducibility/MxN reporting

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
We're continuing to build out our MxN reproducibility reports to assess inter- and intra-day variation, responding to user requests

#### Related Pull Requests
* https://github.com/LabKey/commonAssays/pull/328
* https://github.com/LabKey/platform/pull/2239

#### Changes
* Show peptide sequence coverage along with protein features
* Combine annotations and metadata sections
* Easy download link for Skyline documents in more detail pages
* Combine precursor listings, show filtering results based on sequence length
* Use matrix layout for chromatograms
* Improvements for comparison plots and export support

### CODE REVIEW

File: src/org/labkey/targetedms/TargetedMSController.java [hunk 1]
```diff
@@ -4547,6 +4524,63 @@ public void addNavTrail(NavTree root)
...
+            List<IKeyword> keywords = TargetedMSManager.getKeywords(seqId);
+
+            List<HtmlString> biologicalProcesses = new ArrayList<>();
+            List<HtmlString> molecularFunctions = new ArrayList<>();
+
+            for (IKeyword keyword : keywords)
+            {
+                HtmlString link = new Link.LinkBuilder(keyword.label).href("https://www.uniprot.org/keywords/" + keyword.id).target("_blank").clearClasses().getHtmlString();
+                if (IKeyword.BIOLOGICAL_PROCESS_CATEGORY.equals(keyword.categoryId))
+                    biologicalProcesses.add(link);
```
Comment (labkey-ankurj) [inline src/org/labkey/targetedms/TargetedMSController.java]:
```suggestion
                    extraAnnotations.computeIfAbsent("Biological Processes", k -> new List<HtmlString>()).add(link);
```

File: src/org/labkey/targetedms/TargetedMSController.java [hunk 2]
```diff
@@ -4547,6 +4524,63 @@ public void addNavTrail(NavTree root)
...
+                if (IKeyword.MOLECULAR_FUNCTION_CATEGORY.equals(keyword.categoryId))
+                    molecularFunctions.add(link);
```
Comment (labkey-ankurj) [inline src/org/labkey/targetedms/TargetedMSController.java]:
```suggestion
                    extraAnnotations.computeIfAbsent("Molecular Functions", k -> new List<HtmlString>()).add(link);
```

File: webapp/passport/js/MxN/protein.js [hunk 3]
```diff
@@ -192,188 +74,212 @@ protein =
...
+                PrecursorId: precursor.PrecursorId,
+                ChromatogramId: precursor.PrecursorChromInfoId,
+                PeptideId: precursor.PeptideId,
+                CalibrationCurveId: precursor.CalibrationCurveId,
+
+                peptideSequence: precursor.PeptideSequence,
+                modifiedSequence: precursor.ModifiedSequence,
+                mz: precursor.Mz,
+                peptideId: precursor.PeptideId,
+                precursorId: precursor.PrecursorId,
```
Comment (labkey-ankurj) [inline webapp/passport/js/MxN/protein.js]: there is PrecursorId present at L181

**Review by labkey-ankurj — APPROVED** [review]
minor feedback.

File: webapp/passport/js/MxN/protein.js (hunk 3 above)
Comment (labkey-jeckels) [reply webapp/passport/js/MxN/protein.js]: Thanks for the reminder. I collapsed two different data structures into one and failed to clean up the duplicates and casing. Now consistent and de-duped

**Pull request merged**